components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -299,6 +299,29 @@ void EnsureRewardsBaseDirectoryExists(const base::FilePath& path) {
     base::CreateDirectory(path);
 }
 
+bool RemoveDataOnFileTaskRunner(
+    PublisherInfoDatabase* backend,
+    int32_t remove_mask,
+    uint64_t reconcile_stamp) {
+  bool success = false;
+  if (remove_mask & ledger::ClearDataTypes::DELETE_OTHER_DATA) {
+    success = backend->RemoveAllEntries();
+    backend->Vacuum();
```

Comment:
why does this vacuum run here instead of after both?

Reply:
Fixed